src/xo-app/settings/remotes/index.js

```diff
@@ -39,6 +42,31 @@ class AbstractRemote extends Component {
     return editRemote(remote, {name})
   }
 
+  _test = () => {
+    const { remote } = this.props
+    testRemote(remote).then(answer => {
+      const title = <span>
+        <Icon icon={answer.success ? 'success' : 'error'} />
+        {' '}
+        {answer.success ? _('remoteTestSuccess') : _('remoteTestFailure')} {remote.name}
+      </span>
+      let body = ''
+      if (!answer.success) {
+        body = <p>
+          <dl className='dl-horizontal'>
+            <dt>{_('remoteTestError')}</dt>
+            <dd>{answer.error}</dd>
+            <dt>{_('remoteTestStep')}</dt>
+            <dd>{answer.step}</dd>
+            <dt>{_('remoteTestFile')}</dt>
+            <dd>{answer.file}</dd>
```

Comment:
I see. Let's keep the server side like it is for now, we'll discuss possible improvements later :)